src/main/java/core/mate/academy/service/MachineProducer.java

```diff
@@ -1,11 +1,12 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Machine;
 import java.util.List;
 
-public interface MachineProducer {
+public interface MachineProducer<T extends Machine> {
     /**
      * In your implementations - create 2-3 models, add them to the list and return
      * @return - the list of models
      */
-    List<Object> get();
+    List<? extends Machine> get();
```

Comment:
The `MachineProducer` interface's `get()` method should be parameterized with the generic type parameter `T` to ensure that any implementation is using `Machine` or its subtype. The current signature allows for raw usage which can lead to unchecked assignment warnings and should be avoided. Consider changing the return type to `List<T>`.